### ISSUE TITLE
508 Issues from pa11y-rails

### ISSUE THREAD

**Issue by nickbristow** [18F/forecast#62]
Results can be recreated using [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/)
## Results for https://forecast-staging.18f.gov
- **Error** Form does not contain a submit button (input type="submit", input type="image", or button type="submit").
  - WCAG2AA.Principle3.Guideline3_2.3_2_2.H32.2
  - `#results > form`
  - `<form class="usa-search usa-search-big">
    <div role="search">
      ...</form>`
- **Error** This button element does not have a name available to an accessibility API. Valid names are: title attribute, element content.
  - WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Button.Name
  - `#results > div > drawer > button`
  - `<button class="fa fa-close" data-drawer-toggle="close" tabindex="-1"></button>`
- **Error** Anchor element found with a valid href attribute, but no link content has been supplied.
  - WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
  - `html > body > footer > div > div > div:nth-child(2) > div > a:nth-child(1)`
  - `<a href="#">
            <span class="fa-...</a>`
- **Error** Anchor element found with a valid href attribute, but no link content has been supplied.
  - WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
  - `html > body > footer > div > div > div:nth-child(2) > div > a:nth-child(2)`
  - `<a href="https://twitter.com/GSAOSBU">
            <span class="fa-...</a>`
- **Error** Anchor element found with a valid href attribute, but no link content has been supplied.
  - WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
  - `html > body > footer > div > div > div:nth-child(2) > div > a:nth-child(3)`
  - `<a href="https://www.youtube.com/playlist?list=PLvdwyPgXnxxVYR0faPu0NJ-dIXUqIVsUW">
            <span class="fa-...</a>`
- **Error** Anchor element found with a valid href attribute, but no link content has been supplied.
  - WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
  - `html > body > footer > div > div > div:nth-child(2) > div > a:nth-child(4)`
  - `<a href="#">
            <span class="fa-...</a>`
- **Error** Duplicate id attribute value "opportunity-row" found on the web page.
  - WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
  - `#opportunity-row`
  - `<li class="opportunity-row" id="opportunity-row">
    <div class="office">CFO-Chie...</li>`
- **Error** The html element should have a lang or xml:lang attribute which describes the language of the document.
  - WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
  - `html`
  - `<html><head>
    <title>
    OSBU Forec...</html>`
- **Error** Form does not contain a submit button (input type="submit", input type="image", or button type="submit").
  - WCAG2AA.Principle3.Guideline3_2.3_2_2.H32.2
  - `#agency`
  - `<form id="agency">
      <div class="form-group...</form>`
### Summary

**Errors:** 9

**Comment by gemfarmer** [comment]
@nickbristow I address these issues in #68. Once those changes are in, will the site be considered compliant, or do I also need to address the HTML Code Sniffer's warnings and notices?

**Comment by nickbristow** [comment]
@gemfarmer 
so the warnings and notices just mean they need to be checked by hand. 

I just took a quick look, the main thing I'm seeing that wasn't picked up by the automated scan is the filters on the left side of the page.

They aren't accessible via the keyboard because they have a `tabindex='-1'`. 

That needs to be fixed. And once the PR goes through, go ahead and ping me and I can do a full manual review.

**Comment by gemfarmer** [comment]
Ok. Will do. Thanks @nickbristow 

Also, it looks like much of the _by hand work_ is also intuitive heading size ordering. I might need to chat with you on Monday to gauge what is considered acceptable.

**Comment by gemfarmer** [comment]
@nickbristow I think this is ready for a more thorough audit. I still haven't addressed the tabbing issue you mentioned above
